### PR TITLE
feat(remix): Support new redirect urls as environment variables or options to rootAuthLoader

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

As options
- signInForceRedirectUrl
- signUpForceRedirectUrl
- signInFallbackRedirectUrl
- signUpFallbackRedirectUrl

As environment variables
- CLERK_SIGN_IN_FORCE_REDIRECT_URL
- CLERK_SIGN_UP_FORCE_REDIRECT_URL
- CLERK_SIGN_IN_FALLBACK_REDIRECT_URL
- CLERK_SIGN_UP_FALLBACK_REDIRECT_URL

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/remix/src/ssr/rootAuthLoader.ts [hunk 1]
```diff
@@ -49,7 +49,8 @@ export const rootAuthLoader: RootAuthLoader = async (
 
   const loadedOptions = loadOptions(args, opts);
   // Note: authenticateRequest() will throw a redirect if the auth state is determined to be handshake
-  const requestState = await authenticateRequest(args, loadedOptions);
+  const _requestState = await authenticateRequest(args, loadedOptions);
+  const requestState = { ...loadedOptions, ..._requestState };
```
Comment (panteliselef) [inline packages/remix/src/ssr/rootAuthLoader.ts:53]: Should @clerk/backend be aware of these new redirect urls ? Seems like `afterXUrl` are being returned as part of the SignedIn/OutState

Comment (dimkl) [reply packages/remix/src/ssr/rootAuthLoader.ts:53]: ❓ Aren't those properties only available in the client? I believe that the only reason the `afterXUrl` exist in the backend is for debugging.
@nikosdouvlis Do you have any more context?

Comment (panteliselef) [reply packages/remix/src/ssr/rootAuthLoader.ts:53]: Yes, i would expect them to only be used in the client as well.
Do you think we should update add the new urls in the request state for usage in debugging as well ? Or is it ok to leave the PR as it is now ?

File: packages/remix/src/client/RemixClerkProvider.tsx [hunk 2]
```diff
@@ -80,6 +84,10 @@ export function ClerkProvider({ children, ...rest }: ClerkProviderPropsWithState
     signUpUrl: __signUpUrl,
     afterSignInUrl: __afterSignInUrl,
     afterSignUpUrl: __afterSignUpUrl,
```
Comment (octoper) [inline packages/remix/src/client/RemixClerkProvider.tsx:86]: ❓ Do we still need these or can we mark them as deprecated?

File: packages/remix/src/ssr/rootAuthLoader.ts (hunk 1 above)
Comment (nikosdouvlis) [reply packages/remix/src/ssr/rootAuthLoader.ts:53]: Why wouldn't we want them in `loadedOptions`? I think we can safely add them - improved debuggability is a welcome side effect

**Review by nikosdouvlis — APPROVED** [review]

**Pull request merged**